contrib/slurm/slurmexecutor.py

```diff
@@ -0,0 +1,297 @@
+                    os.rename(run.log_file, run.log_file + ".killed")
+                else:
+                    os.remove(run.log_file)
+            except OSError:
+                pass
+            return 1
+
+        run.set_result(run_result)
+        self.output_handler.output_after_run(run)
+        return None
+
+
+def run_slurm(benchmark, args, log_file, timelimit, cpus, memory):
+    srun_timelimit_h = int(timelimit / 3600)
+    srun_timelimit_m = int((timelimit % 3600) / 60)
+    srun_timelimit_s = int(timelimit % 60)
+    srun_timelimit = f"{srun_timelimit_h}:{srun_timelimit_m}:{srun_timelimit_s}"
+
+    mem_per_cpu = int(memory / cpus / 1000000)
+
+    assert benchmark.config.scratchdir and os.path.exists(benchmark.config.scratchdir)
+
+    with tempfile.TemporaryDirectory(dir=benchmark.config.scratchdir) as tempdir:
+
+        os.makedirs(os.path.join(tempdir, "upper"))
+        os.makedirs(os.path.join(tempdir, "work"))
+
+        tool_command = " ".join(args)
+        singularity_command = (
+            f"singularity exec "
+            f"-B $PWD:/lower --no-home "
+            f"-B {tempdir}:/overlay "
+            f'--fusemount "container:fuse-overlayfs -o lowerdir=/lower -o upperdir=/overlay/upper -o workdir=/overlay/work $HOME" '
+            f"{benchmark.config.singularity} {tool_command}"
+            if benchmark.config.singularity
+            else tool_command
+        )
+        srun_command = (
+            f"srun "
+            f"-t {srun_timelimit} "
+            f"-c {cpus} "
+            f"-o {log_file} "
+            f"--mem-per-cpu {mem_per_cpu} "
+            f"--threads-per-core=1 "
+            f"--ntasks=1 "
+            f"{singularity_command}"
+        )
+        jobid_command = (
+            f"{srun_command} 2>&1 | grep -o 'job [0-9]* queued' | grep -o '[0-9]*'"
+        )
+        seff_command = f"seff $({jobid_command})"
+        logging.debug("Command to run: %s", seff_command)
+        result = subprocess.run(
+            ["bash", "-c", seff_command], shell=False, stdout=subprocess.PIPE
+        )
+
+    # Runexec would populate the first 6 lines with metadata
+    with open(log_file, "r+") as file:
+        content = file.read()
+        file.seek(0, 0)
+        empty_lines = "\n" * 6
+        file.write(empty_lines + content)
+
+    exit_code, cpu_time, wall_time, memory_usage = parse_seff(str(result.stdout))
+
+    return {
+        "starttime": benchexec.util.read_local_time(),
+        "walltime": wall_time,
+        "cputime": cpu_time,
+        "memory": memory_usage,
+        "exitcode": ProcessExitCode(raw=exit_code, value=exit_code, signal=None),
+    }
+
+
+def parse_seff(result):
+    exit_code_pattern = re.compile(r"State: COMPLETED \(exit code (\d+)\)")
+    cpu_time_pattern = re.compile(r"CPU Utilized: (\d+):(\d+):(\d+)")
+    wall_time_pattern = re.compile(r"Job Wall-clock time: (\d+):(\d+):(\d+)")
+    memory_pattern = re.compile(r"Memory Utilized: (\d+\.\d+) MB")
```

Comment:
Here you use compilation of regexps to speed it up, but you do this for every run. Either the compilation is irrelevant because the `re` module caches them internally anyway, or you want to do this in a module constant such that the regexp instances are reused for all runs.